pkg/cache/cache.go

```diff
@@ -221,6 +239,20 @@ func (c *Cache) ClusterQueueTerminating(name string) bool {
 	return c.clusterQueueInStatus(name, terminating)
 }
 
+func (c *Cache) ClusterQueueReadiness(name string) (metav1.ConditionStatus, string, string) {
+	c.RLock()
+	defer c.RUnlock()
+	cq := c.clusterQueues[name]
+	if cq == nil {
+		return metav1.ConditionFalse, "NotFound", "Cluster queue not found"
+	}
+	if cq != nil && cq.Status == active {
+		return metav1.ConditionTrue, "Ready", "Can admit new workloads"
+	}
+	reason, msg := cq.inactiveReason()
+	return metav1.ConditionFalse, reason, msg
```

Comment:
Uhm, this function's semantic looks like overwrapping the `ClusterQueueActive` function.
In the future, it would be better to refactor the whole of the cache or consolidate `func ClusterQueueActive` with this function.

Reply:
@stuton can you remove the existing ClusterQueueActive function if it's no longer used?

Reply:
`ClusterQueueActive` is used and usually in places  where the inactivity reason is not relevant. We could replace it's usage with `ClusterQueueReadiness` but the time spent on getting the exact reason will be wasted.

Reply:
Ack. Please attend to the other comments.